src/map.c

```diff
@@ -222,20 +230,24 @@ static void Map_DecreaseStorageKeysValues(MAP_HANDLE_DATA* handleData)
     {
         /*certainly > 1...*/
         char** undoneValues;
-        char** undoneKeys = (char**)realloc(handleData->keys, sizeof(char*)* (handleData->count - 1));
-        if (undoneKeys == NULL)
+        char** undoneKeys;
+        size_t realloc_size = safe_subtract_size_t(handleData->count, 1);
+        realloc_size = safe_multiply_size_t(realloc_size, sizeof(char*));
+
+        if (realloc_size == SIZE_MAX ||
+            (undoneKeys = (char**)realloc(handleData->keys, realloc_size)) == NULL)
         {
-            LogError("CATASTROPHIC error, unable to undo through realloc to a smaller size");
+            LogError("CATASTROPHIC error, unable to undo through realloc to a smaller size, size:%zu", realloc_size);
         }
         else
         {
             handleData->keys = undoneKeys;
         }
 
-        undoneValues = (char**)realloc(handleData->values, sizeof(char*)* (handleData->count - 1));
-        if (undoneValues == NULL)
+        if (realloc_size == SIZE_MAX || 
```

Comment:
Looks like this condition was already checked on line 237

Reply:
yea, there is no function exit on the first one.